Menu/LobbySelectMenu.cs

```diff
@@ -260,6 +309,70 @@ public override void Update()
             // todo out of view
         }
 
+        private void FilterLobbyCards()
+        {
+            var targetLobbies = new List<LobbyInfo>();
+
+            for (int i = 0; i < lobbies.Length; i++)
+            {
+                var lobby = lobbies[i];
+
+                if (filter.lobbyName != "" && !lobby.name.ToLower().Contains(filter.lobbyName)) continue;
+                if (filter.mode != "All" && lobby.mode != filter.mode) continue;
+                if (filter.password == "NoPassword" && lobby.hasPassword) continue;
+                if (filter.password == "Password" && !lobby.hasPassword) continue;
+                if (filter.playerLimit < lobby.maxPlayerCount) continue;
+
+                targetLobbies.Add(lobby);
+            }
+
+            displayLobbies = targetLobbies.ToArray();
+        }
+
+        private void UpdateLobbyFilter()
+        {
+            filter.mode = filterModeDropDown.value;
+            filter.password = filterPasswordDropDown.value;
+            filter.playerLimit = filterLobbyLimit.valueInt;
+            filter.lobbyName = lobbySearchInputBox.value.ToLower();
+
+            CreateLobbyCards();
+        }
+
+        private class LobbyFilter
+        {
+            public bool enabled;
+            public string mode;
+            public string password;
+            public int playerLimit;
+            public string lobbyName;
+            public enum ModeFilter
+            {
+                All,
+                Meadow,
+                Story,
+                ArenaCompetitive
+            };
+            public enum PasswordFilter
+            {
+                [Description("Don't care")]
```

Comment:
Update these please so we can have the game _attempt_ to benefit translating them like pictured below:

![image](https://github.com/user-attachments/assets/3876dd00-9ae0-4574-b1ac-7d8431657f5c)

Reply:
![image](https://github.com/user-attachments/assets/c1edb9fe-b58a-4ac9-9bc9-af455db2e903)

like so correct?